### ISSUE TITLE
cold-start tasks causing downstream graphing to become immortal

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc#2090]
Since 6.11.0 cold-start tasks can cause downstream graphing to become immortal. Possibly related to #1773.

Example suite:
```
[scheduling]
    initial cycle point = 20000101T00
    [[special tasks]]
        cold-start = housekeep_cold
    [[dependencies]]
        [[[P1D]]]
            graph = housekeep_cold | housekeep[-P1D] => some_task => housekeep
```

**Comment by arjclark** [comment]
For clarity, the downstream tasks themselves hang around in the task pool forever and don't clear out.

**Comment by hjoliver** [comment]
Closed by #2091
